### ISSUE TITLE
freezing motion video

### ISSUE THREAD

**Issue by funnyQZQ** [google-research/mint#9]
@liruilong940607 Hi, Ruilong, i run the evaluation code with the pretrained model you provided on google drive, however the  result video like [this](https://user-images.githubusercontent.com/25003964/134494526-1bd39591-6776-4f06-a927-63cf2963dde5.mp4), the first two seconds motion from the seed motion, and then got freezing motion in the following seconds. I don't know why. Could you please give me some suggestions?

**Comment by oxygenman** [comment]
same problem,after 120 frams,the range of motion becomes smaller,some times it keep still, is there need to set some parameters? @liruilong940607

**Comment by tosaka0822** [comment]
where is the evaluation code can result video?
only python tools/calculate_fid_scores.py 
100%|████████████████████████████████| 268/268 [14:10<00:00,  3.17s/it]
Evaluation: FID_k: 33566817085848.8477, FID_g: 190.4702
Evaluation: Dist_k: 138815.0493, Dist_g: 3.4433

**Comment by funnyQZQ** [comment]
You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).

**Comment by caijianfei** [comment]
same problem, is there any suggestion to solve the problem ?

**Comment by funnyQZQ** [comment]
@shanyangmie Any update for this issue?

**Comment by shanyang-me** [comment]
Please make sure that the parameters are set correctly, otherwise this shouldn't be a problem.

**Comment by caijianfei** [comment]
@shanyang-me many thx for your feedback, I used audio feature from cached features output by "tools/preprocessing.py" (aist_audio_feats/mBR0.npy) with 3241 frames, and used first 120 frames from motion annotation data (annotations/motions/gBR_sBM_cAll_d04_mBR0_ch01.pkl) as motion seed, but the output motion still freezing after about 300-400 frames, my code and generated motion can be gotten from https://github.com/caijianfei/google-dancing, Is there any wrong setting?  I would appreciate it if you could give me some suggestions, thx very much

**Comment by YiAthena** [comment]
I met the same problem as @caijianfei .

**Comment by Concernin9** [comment]
> You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).

Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!

**Comment by Concernin9** [comment]
> You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).

@NewCoderQ  Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!

**Comment by Concernin9** [comment]
> @shanyang-me many thx for your feedback, I used audio feature from cached features output by "tools/preprocessing.py" (aist_audio_feats/mBR0.npy) with 3241 frames, and used first 120 frames from motion annotation data (annotations/motions/gBR_sBM_cAll_d04_mBR0_ch01.pkl) as motion seed, but the output motion still freezing after about 300-400 frames, my code and generated motion can be gotten from https://github.com/caijianfei/google-dancing, Is there any wrong setting? I would appreciate it if you could give me some suggestions, thx very much

@caijianfei Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!

**Comment by Concernin9** [comment]
> where is the evaluation code can result video? only python tools/calculate_fid_scores.py 100%|████████████████████████████████| 268/268 [14:10<00:00, 3.17s/it] Evaluation: FID_k: 33566817085848.8477, FID_g: 190.4702 Evaluation: Dist_k: 138815.0493, Dist_g: 3.4433

@tosaka0822  Hello, have you solved this problem?

**Comment by funnyQZQ** [comment]
> > You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).
> 
> @NewCoderQ Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!

No, you don't need to convert pose data from .npy to .pkl file, you can read the pose and translation data from the .npy file, and inference with SMPL model.

**Comment by YiAthena** [comment]
hi @NewCoderQ I found It could be slightly better if set the bpm to 120.

**Comment by JingwenLiang** [comment]
> 

Are we suppose to use this command in AIST++ api to visulize the result?
```
# install some additional libraries for 3D mesh visualization
pip install vedo trimesh

python demos/run_vis.py \
  --anno_dir <ANNOTATIONS_DIR> \
  --smpl_dir <SMPL_DIR> \
  --video_name gWA_sFM_c01_d27_mWA2_ch21 \ 
  --mode SMPLMesh```

Can we get some instructions on how to read the result .npy file

**Comment by h310558606** [comment]
> > You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).
> 
> @NewCoderQ Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!
Hello, have you solved this problem?

**Comment by YiAthena** [comment]
hi @NewCoderQ i trained from scratch and solved the freezing problem.

**Comment by h310558606** [comment]
Hi,
Could you tell me how to transfer the npy file to the bvh file?
The bvh_writer.py can not work at all.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "google-research/mint"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月25日(星期五) 下午5:09
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [google-research/mint] freezing motion video (#9)





 
hi @NewCoderQ i trained from scratch and solved the freezing problem.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by funnyQZQ** [comment]
> hi @NewCoderQ i trained from scratch and solved the freezing problem.

Well done! I will have a try.

**Comment by caijianfei** [comment]
> hi @NewCoderQ i trained from scratch and solved the freezing problem.

Great!! How long did u take to retrain the model ?

**Comment by miibotree** [comment]
> hi @NewCoderQ i trained from scratch and solved the freezing problem.

I retrained the model from scratch, changing batch size to 64 and remain other parameters unchanged. After 7W steps the loss seems convergence, I stopped and checked the result.However the result is still not good, seems collapse after 2 sec.

**Comment by hzy-del** [comment]
> > You need to visualize the predicted dance motion like [this](https://github.com/google/aistplusplus_api/blob/091f00b24505d0b6635ffb2cd0eaeccdfda4ed86/README.md#visualize-the-smpl-mesh).
> 
> @NewCoderQ Hello, the result of the code is the data in NPY format. I found that to form an smpl model, do I need to convert NPY format into PKL format first? I want to visualize the results. Is this the process? Looking forward to your reply, thank you very much!

hello，have you solved the problem?
